### PR TITLE
[LogisticRegression] Use cupy to standardize dense vectors for better numeric stability when standardization is on

### PR DESCRIPTION
**Pull request by lijinf2**

### CODE REVIEW

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/classification.py [hunk 1]
```diff
@@ -976,14 +978,14 @@ def _logistic_regression_fit(
...
-            if self.getStandardization() is True and is_sparse is True:
+            if standardization is True and is_sparse is True:
                 concated = concated.toarray()
 
             # Padding zero columns so that the total number of columns is a multiple of 32.
             # The purpose is to avoid out-of-bound memory access of some RAFT 24.02 cuda kernels (e.g. sum).
             # This is a temporary workaround and is expected to be removed in 24.04
             num_pad_zero_cols = 0
-            if self.getStandardization() is True and concated.shape[1] % 32 != 0:
+            if standardization is True and concated.shape[1] % 32 != 0:
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/classification.py:988]: Ok to skip the padding since problematic steps being bypassed?

Comment (lijinf2) [reply python/src/spark_rapids_ml/classification.py:988]: I see your point. Yes, it was for standardization only and is not required after this patch.

**Comment by lijinf2** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**